prism/src/main/java/com/hipo/android/prism/Prism.java

```diff
@@ -0,0 +1,117 @@
+package com.hipo.android.prism;
+
+public class Prism {
+
+    private static final String QUESTION_MARK = "?";
+
+    private static final String WIDTH_QUERY = "&width=";
+    private static final String HEIGHT_QUERY = "&height=";
+    private static final String CROP_X = "&crop_x=";
+    private static final String CROP_Y = "&crop_y=";
+    private static final String CROP_WIDTH = "&crop_width=";
+    private static final String CROP_HEIGHT = "&crop_height=";
+
+    private static volatile Prism singleton = null;
+
+    private String url;
+    private Integer width;
+    private Integer height;
+    private Integer cropx;
+    private Integer cropy;
+    private Integer cropWidth;
+    private Integer cropHeight;
+
+    public static Prism withUrl(String url) {
+        if (singleton == null) {
+            synchronized (Prism.class) {
+                if (singleton == null) {
+                    singleton = new Builder(url + QUESTION_MARK).build();
+                }
+            }
+        }
+        return singleton;
+    }
+
+    public Prism width(Integer width) {
+        singleton.width = width;
+        return singleton;
+    }
+
+    public Prism height(Integer height) {
+        singleton.height = height;
+        return singleton;
+    }
+
+    public Prism cropX(Integer cropx) {
+        singleton.cropx = cropx;
+        return singleton;
+    }
+
+    public Prism cropY(Integer cropy) {
+        singleton.cropy = cropy;
+        return singleton;
+    }
+
+    public Prism cropWidth(Integer cropWidth) {
+        singleton.cropWidth = cropWidth;
+        return singleton;
+    }
+
+    public Prism cropHeight(Integer cropHeight) {
+        singleton.cropHeight = cropHeight;
+        return singleton;
+    }
+
+    Prism(String url) {
+        this.url = url;
+    }
+
+    public static class Builder {
+        private final String url;
+
+        /** Start building a new {@link Prism} instance. */
+        public Builder(String url) {
+            if (url == null) {
+                throw new IllegalArgumentException("Context must not be null.");
```

Comment:
We will check url empty or null, and we need the throw string exception not context